packages/core/src/enqueue_links/enqueue_links.ts

```diff
@@ -123,43 +138,74 @@ export async function enqueueLinks(options: EnqueueLinksOptions): Promise<QueueO
         baseUrl: ow.optional.string,
         pseudoUrls: ow.any(ow.null, ow.optional.array.ofType(ow.any(
             ow.string,
-            ow.regExp,
             ow.object.hasKeys('purl'),
             ow.object.validate(validators.pseudoUrl),
         ))),
+        globs: ow.any(ow.null, ow.optional.array.ofType(ow.string)),
+        regexps: ow.any(ow.null, ow.optional.array.ofType(ow.regExp)),
         transformRequestFunction: ow.optional.function,
         strategy: ow.optional.string.oneOf([
             EnqueueStrategy.All,
             EnqueueStrategy.SameDomain,
         ]),
     }));
 
-    const extraPseudoUrls = [];
+    const {
+        requestQueue,
+        limit,
+        urls,
+        pseudoUrls,
+        globs,
+        regexps: regexpsInput,
+        transformRequestFunction,
+    } = options;
+
+    let regexps: RegExp[] = [];
+
+    if (pseudoUrls?.length) {
+        log.deprecated('`pseudoUrls` option is deprecated, use `globs` or `regexps` instead');
+        regexps = regexps.concat(constructRegExps(pseudoUrls));
+    }
+
+    if (globs?.length) {
+        for (const glob of globs) {
+            const trimmedGlob = glob.trim();
+            if (trimmedGlob.length === 0) throw new Error(`Cannot parse glob pattern '${trimmedGlob}': it must be an non-empty string`);
+
+            const re = makeRe(glob, { nocase: true });
+            if (re) {
+                regexps.push(re);
+            } else {
+                log.warning(`Provided glob pattern '${glob}' is invalid. Pattern will be skipped.`);
+            }
+        }
+    }
+
+    if (regexpsInput?.length) {
+        regexps.concat(regexpsInput);
+    }
 
     if (options.baseUrl) {
         switch (options.strategy ?? EnqueueStrategy.SameDomain) {
             case EnqueueStrategy.SameDomain:
                 // We need to get the origin of the passed in domain in the event someone sets baseUrl
                 // to a url like https://example.com/deep/default/path and one of the found urls is an
                 // absolute relative path (/path/to/page)
-                extraPseudoUrls.push(`${new URL(options.baseUrl).origin}/[.*]`);
+                regexps.push(new RegExp(new URL(options.baseUrl).origin));
```

Comment:
This won't match the same although it might be fine? or maybe we should specifically make it match the start of the string?

Reply:
yeah this needs to be definitly fixed, you can use `purlToRegExp` to build the regexp and inline it here